lib/samson/secrets/vault_client_manager.rb

```diff
@@ -105,16 +116,16 @@ def responsible_clients(id)
 
         if deploy_group_permalink == 'global'
           if environment_permalink == 'global'
-            clients.values
+            clients.values.presence || raise(VaultServerNotConfigured, "no vault servers configured")
           else
-            unless deploy_group_permalinks = client_map[:environments][environment_permalink]
-              raise "no environment with permalink #{environment_permalink} found"
+            unless deploy_group_permalinks = client_map[:environments][environment_permalink].presence
+              raise VaultServerNotConfigured, "no environment with permalink #{environment_permalink} found"
             end
             deploy_group_permalinks.map { |p| client(p) }.uniq
           end
         else
           [client(deploy_group_permalink)]
-        end.presence || raise("no vault servers found for #{id}")
+        end
```

Comment:
each branch now takes care of missing servers so we never have to say "idk what happened"